src/gt4py/next/program_processors/runners/dace/gtir_sdfg.py

```diff
@@ -503,8 +501,11 @@ def make_temps(
             else:
                 temp, _ = self.add_temp_array_like(sdfg, desc)
                 temp_node = head_state.add_access(temp)
+                field_range = dace.subsets.Range.from_array(desc)
                 head_state.add_nedge(
-                    field.dc_node, temp_node, sdfg.make_array_memlet(field.dc_node.data)
+                    field.dc_node,
+                    temp_node,
+                    dace.Memlet(data=temp, subset=field_range, other_subset=field_range),
```

Comment:
```suggestion
                    dace.Memlet(data=temp, subset=temp_range, other_subset=field_range),
```

- There was some kind of sharing between the subsets, if you modify one it would also change the other, which is dangerous (it is essentially like calling `copy()` on a `list` of NumPy objects and then modify the objects in the copy).
- Furthermore this Memlet is highly non trivial as `data` refers to the destination, so giving better names to the argument is better.